django_app/redbox_app/settings.py

```diff
@@ -156,19 +156,20 @@
 CSP_SCRIPT_SRC = (
     "'self'",
     "'sha256-GUQ5ad8JK5KmEWmROf3LZd9ge94daqNvd8xy9YS1iDw='",
+    "'sha256-TnW3b3osC+i1wd8tjlnl6QJOeYc4oc7edkmhziTLEfk='", # Mermaid
     "plausible.io",
     "eu.i.posthog.com",
     "eu-assets.i.posthog.com",
 )
 CSP_OBJECT_SRC = ("'none'",)
 CSP_REQUIRE_TRUSTED_TYPES_FOR = ("'script'",)
-CSP_TRUSTED_TYPES = ("dompurify", "default")
+CSP_TRUSTED_TYPES = ("dompurify", "default", "'allow-duplicates'")
 CSP_REPORT_TO = "csp-endpoint"
 CSP_FONT_SRC = (
     "'self'",
     "s3.amazonaws.com",
 )
-CSP_STYLE_SRC = ("'self'",)
+CSP_STYLE_SRC = ("'self'", "'unsafe-inline'") # TO DO: Not happy about this change - look at workarounds (KE)
```

Comment:
Thanks Will! I'm going to try a JavaScript workaround in the first instance, as that will be the quickest solution (if it works!).

I think the SSR approach is also a strong contender. A little more fiddly because of the streaming, but I can see a couple of ways of approaching that.

Any thoughts/concerns around the system prompts?

Reply:
We could add something to the effect of "if asked to create a diagram use the mermaid markdown format". I find that pretty much every LLM is quite fluent in it